specification/_types/aggregations/metric.ts

```diff
@@ -344,6 +344,11 @@ export class TopHitsAggregation extends MetricAggregationBase {
    * @server_default false
    */
   explain?: boolean
+  /**
+   * Array of wildcard (*) patterns. The request returns values for field names
+   * matching these patterns in the hits.fields property of the response.
+   */
+  fields?: (FieldAndFormat | Field)[]
```

Comment:
should it be `FieldAndFormat[] | Field[]`?

Reply:
That's a good question. My assumption was that `fields` could take an array that consisted of either, or a combination of the two (e.g. `['field_one', { field: 'field_two', format: 'xxx' }]`). [The docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-fields.html#search-fields-request) suggest that's the case as well.

I was looking at the `FieldFetcher` subphase in the Elasticsearch repo to verify in the source, but my Java is pretty rusty so it's hard to tell. What does it seem like to you?